### ISSUE TITLE
Sporadic load timeout in command set extensions that stop extension from loading

### ISSUE THREAD

**Issue by hesperanca** [SharePoint/sp-dev-docs#8970]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- Edge - 113.0.1774.35
- Chrome - 113.0.5672.93
- SPFx 1.17.2 (but the problem was also happening in previous versions)
- Node.js version - 16.20.0

### Describe the bug / error

Hi,

We've been having problems with command set extensions sporadically falling to load for a while. Doing a full page refresh (CTRL+F5) always fixes the issue but our customers are getting frustrated by the experience (we are an ISV).

In our case we have a command set extension that simply displays a button in the top menu. The option should always be visible (independently if we have items selected or not). The extension is deployed to a site collection app catalogue and hosted in SharePoint (not using a CDN). 

We noticed that, when the user opens a new browser window and navigates to the library for the first time, the menu button does not show and after a few seconds we see the timeout error messages in dev tools (see screenshot). If the user does a full page refresh the button starts to display again and no errors are generated.

In total we have one command set extension and two application customisers on the page that generate the error although the application customisers do not seem to be affected by it.

The module that seems to be falling most of the time is the module with our localisation strings (the application customisers and the command set extension all use the same module with the localisation strings).

Any help would be greatly appreciated:

![image](https://github.com/SharePoint/sp-dev-docs/assets/29897247/80646170-7c8d-4637-aad1-14c5c18248b7)

### Steps to reproduce

1. Open the browser window and navigate to a SharePoint document library where the command set extension is installed
2. Command set option does not show in about 80% of the cases
3. Doing a full refresh the command set option starts to show

### Expected behavior

Command set option should be displayed consitently

**Comment by AJIXuMuK** [comment]
@hesperanca - thanks for submitting the issue!
I'm seeing that in both app customizer and command set errors the problem is with loading localization strings from the same module.
Do you have a library component with a localization that is used by both extensions?
Could you please check if the id `bddbicc5-2a94-4cb8-9cc2-dfcda2ea6353` is for the app customizer, command set or any other component?
And is `0.0.28` the correct version of it?

**Comment by AJIXuMuK** [comment]
@hesperanca - I'm sorry to hear that.
Could you please make me a favor and share few things with me?
After you bundle and package solution, you should have a `release` folder.
Could you please share with me the json from `component-dependency-audit` folder as well as the generated manifest from `manifests` folder (both folders are under `release`).
Also would you mind opening `ClientSideAssets` list in your app catalog site (https://<your-tenant>.sharepoint.com/sites/<your-app-catalog>/ClientSideAssets), open the folders with ids of your extensions and check if you're seeing the localization js file in there.

I'm a bit concerned here as all your solutions are trying to load the loc file from the same module that should be a library component in that case. But you're saying it is not.

**Comment by AJIXuMuK** [comment]
@hesperanca - we have made one change that potentially could fix this issue.
Could you please check on your end?

**Comment by hesperanca** [comment]
Hi Alex, thanks for your quick reply and your help in trying to sort this issue.

I can't seem to be able to recreate the issue this morning (we are based in the UK). How do I have access to the changes that you made? Do I need to download and install a new SPFX version or did you change something server-side (which would explain why I can't recreate the issue)?

We spent a substantial amount of time investigating the issue yesterday morning. As you mentioned, the problem seems to be related to sharing the localization strings module. FYI, our solution is quite big, with a lot of components that are shared between extensions and webparts. We centralised all the localization strings instead of having one file per extension/webpart. We followed the instructions from this blog post to centralise the localization strings in one file: https://www.eliostruyf.com/simplify-localization-in-sharepoint-framework-projects/

During our tests yesterday we managed to recreate the issue pretty consistently. Here are the steps we have taken:
1) Load the homepage of the site and confirmed that the GlobalLocalizations strings file is loaded (we have two application customizers that depend on it and we can see the file loaded in the dev tools network tab).

2) Navigate to a list page (where the command set extension should display a button on the toolbar ) and noticed that the command set extension didn't run (we are console logging some messages in the onInit message that were never shown). 

3) Waited for some time and noticed the timeout error messages regarding the localization file. The strange thing is that there were no failed requests showing in the network tab. It's like SP was waiting for the localization file to load but never requested it. I'm assuming that our command set extensions did not run because they have a dependency on the localization file that was never loaded.

As I mentioned I cannot seem to recreate the issue at the moment so I'll keep you updated but let me know if you still want me to send you the information you requested on the previous message.

Thanks for your help so far.
